### ISSUE TITLE
Qualify identifiers using their root module

### ISSUE THREAD

**Issue by silene** [coq/coq#9697]
To avoid potential shadowing issues, Why3 generates Coq scripts that contain fully-qualified identifiers, e.g., `Coq.Init.Datatypes.implb`. This has several disadvantages. The most important ones are as follows.

1. The Why3 scripts are meant for the user, yet the user is basically forced to run Coq (through some IDE) to decipher the generated definitions and theorem statements due to the visual strain caused by overlong identifiers.

2. Fully-qualified names potentially leak implementation details. For example, when `IZR` was moved to an earlier module in Coq 8.7 to fix a dependency issue in the standard library, the generated Why3 scripts broke since the fully qualified name was different. Yet, the only thing that matters is that `IZR` comes from some specific library (the `Coq` one), not that it is located in the submodule `Rdefinitions`.

The `From` vernacular was introduced in 8.5 to fix this issue for required modules. Unfortunately, no corresponding syntax was ever proposed for other identifiers.

To summarize, rather than `Coq.Init.Datatypes.implb`, I would like to be able to type something as short as `Coq..implb`. I don't expect much resistance to the feature itself, but finding a good syntax (i.e., readable yet not conflicting) might prove challenging. Here are some proposals:

- `Coq..implb`
- `Coq.'implb`
- `Coq._.implb`
- `.Coq.implb`

**Comment by gares** [comment]
Recently some "coqlib" cleanup landed in master. It is possible, at the ML side, to use a name like `core.implb` and have in a v file `Register long.long.path core.implb`. In elpi I've added recently a "quotaion" with syntax `lib:core.implb` that is supposed at some point to use the coqlib machinery to expand to the `long.long.path` and make elpi programs more portable across versions.

Would this alternative approach work for why3 too? (after all an indirection can solve all problems in CS, isn't it?)

**Comment by gares** [comment]
I mean, the `lib:core.impl` thing is part of Coq terms, elpi programs contain Coq syntax inside quotations, and I've extended Coq syntax with `lib:...` when elpi is loaded). So you can write something like `{{ fun x : nat => lib:core.nat.add x x }}` to be sure you target precisely `Nat.plus` or whatever it will be called tomorrow.

Maybe this quotation makes sense not only for elpi.
